src/main/java/project/service/UserService.java

```diff
@@ -84,15 +88,49 @@ public User getUser(String userId, String accessTokenFromClient) throws Unauthor
         return user;
     }
 
+
     private void checkIfExistingUserIsAuthorized(final String userId, final String accessTokenFromDatabase,
-                                                 final String accessTokenFromWebPage) throws UnauthorizedException
+                                                 final String accessTokenFromWebPage,
+                                                 final String fbAccessToken) throws UnauthorizedException
     {
-        if (userId == null || userId.isEmpty()
-                || accessTokenFromWebPage == null || accessTokenFromWebPage.isEmpty()
-                || accessTokenFromDatabase == null || accessTokenFromDatabase.isEmpty()
-                || !accessTokenFromDatabase.equals(accessTokenFromWebPage))
+        boolean isUserDataFromDatabaseInvalid = userId == null || userId.isEmpty()
+                || accessTokenFromDatabase == null || accessTokenFromDatabase.isEmpty();
+
+        boolean isAccessTokenFromDatabaseInvalid = accessTokenFromWebPage == null || accessTokenFromWebPage.isEmpty()
+                                                || accessTokenFromDatabase == null
+                                                || !accessTokenFromDatabase.equals(accessTokenFromWebPage);
+
+        if (isUserDataFromDatabaseInvalid)
+        {
+            throw new UnauthorizedException("user is not correct registered in database");
+        }
+        else
         {
-            throw new UnauthorizedException();
+            if (isAccessTokenFromDatabaseInvalid)
+            {
+                // lets check if fb access token is valid
+                if (fbAccessToken == null || fbAccessToken.isEmpty())
+                {
+                    throw new UnauthorizedException("an incorrect access token was sent from the web and the fb access " +
+                            "token was not sent");
+                }
+
+                /* TODO: create call to fb and check if 'fbAccessToken' is valid for userId
```

Comment:
this is TODO,
so you can now login to other user account with fake facebook access token, I am not going to do this until we have decided/changed to new auth flow, see: https://github.com/pajdak3/hermes-server/pull/23/files#r41948046